lib/sisimai/bite/email/einsundeins.rb

```diff
@@ -0,0 +1,143 @@
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+        return nil unless mhead['from']    =~ Re0[:from]
+        return nil unless mhead['subject'] =~ Re0[:subject]
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        v = nil
+
+        hasdivided.each do |e|
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
+              break if blanklines > 1
+              next
+            end
+            rfc822list << e
+
+          else
+            # Before "message/rfc822"
+            next if readcursor & Indicators[:deliverystatus] == 0
+            next if e.empty?
+
+            # The following address failed:
+            #
+            # general@example.eu
+            #
+            # For the following reason:
+            #
+            # Mail size limit exceeded. For explanation visit
+            # http://postmaster.1and1.com/en/error-messages?ip=%1s
+            v = dscontents[-1]
+
+            if cv = e.match(/\A([^ ]+[@][^ ]+)\z/)
+              # general@example.eu
+              if v['recipient']
+                # There are multiple recipient addresses in the message body.
+                dscontents << Sisimai::Bite.DELIVERYSTATUS
+                v = dscontents[-1]
+              end
+              v['recipient'] = cv[1]
+              recipients += 1
+
+            elsif e =~ Re1[:error]
+              # For the following reason:
+              v['diagnosis'] = e
+
+            else
+              # Get error message
+              if v['diagnosis']
+                # Append error message strings
+                v['diagnosis'] += ' ' + e
+              end
+            end
+          end
+        end
+        return nil if recipients.zero?
+        require 'sisimai/string'
+
+        dscontents.map do |e|
+          e['agent']       = self.smtpagent
```

Comment:
__[rubocop]__
* Style/IfInsideElse: Convert `if` nested inside `else` to `elsif`.

<!-- Comment by [SideCI](https://sideci.com) -->